### PR TITLE
ini: Use String for arguments

### PR DESCRIPTION
**Pull request by doctor-rd**

### CODE REVIEW

File: Userland/Utilities/ini.cpp [hunk 1]
```diff
@@ -14,10 +14,10 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
...
     TRY(Core::System::pledge("stdio rpath wpath cpath"));
 
-    char const* path = nullptr;
-    char const* group = nullptr;
-    char const* key = nullptr;
-    char const* value_to_write = nullptr;
+    String path;
+    String group;
+    String key;
+    String value_to_write;
```
Comment (trflynn89) [inline Userland/Utilities/ini.cpp:20]: If we're making this change, why not use `StringView` to avoid allocation?

Comment (trflynn89) [reply Userland/Utilities/ini.cpp:20]: Ah, `ConfigFile::write_entry` would allocate a string anyways. Nevermind.

File: Userland/Utilities/ini.cpp [hunk 2]
```diff
@@ -31,9 +31,9 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return 1;
     }
 
-    auto config = TRY(Core::ConfigFile::open(path, value_to_write ? Core::ConfigFile::AllowWriting::Yes : Core::ConfigFile::AllowWriting::No));
+    auto config = TRY(Core::ConfigFile::open(path, value_to_write.is_empty() ? Core::ConfigFile::AllowWriting::No : Core::ConfigFile::AllowWriting::Yes));
```
Comment (awesomekling) [inline Userland/Utilities/ini.cpp:34]: Doesn't this break the case where you're setting some key to an empty string(`""`)?

Comment (doctor-rd) [reply Userland/Utilities/ini.cpp:34]: The config utility also uses is_empty(). May be this should be fixed.

Comment (awesomekling) [reply Userland/Utilities/ini.cpp]: Yeah, definitely :)

**Pull request merged**